ambry-tools/src/main/java/com.github.ambry/store/DiskReformatter.java

```diff
@@ -0,0 +1,244 @@
+    consistencyChecker =
+        new ConsistencyCheckerTool(clusterMap, new StoreToolsMetrics(clusterMap.getMetricRegistry()), time);
+  }
+
+  /**
+   * Performs a reformat of the disk.
+   * 1. Copies one partition on the disk to a scratch space
+   * 2. Performs local copies of all other partitions on the disk using {@link StoreCopier} and deletes the source.
+   * 3. Copies the partition in the scratch space back on to the disk
+   * 4. Deletes the folder in the scratch space
+   * @param diskMountPath the mount path of the disk to reformat
+   * @param scratch the scratch space to use
+   * @throws Exception
+   */
+  public void reformat(String diskMountPath, File scratch) throws Exception {
+    if (!scratch.exists()) {
+      throw new IllegalArgumentException("Scratch space " + scratch + " does not exist");
+    }
+    List<ReplicaId> replicasOnDisk = new ArrayList<>();
+    // populate the replicas on disk
+    List<? extends ReplicaId> replicaIds = clusterMap.getReplicaIds(dataNodeId);
+    for (ReplicaId replicaId : replicaIds) {
+      if (replicaId.getDiskId().getMountPath().equals(diskMountPath)) {
+        replicasOnDisk.add(replicaId);
+      }
+    }
+    if (replicasOnDisk.size() == 0) {
+      throw new IllegalArgumentException("There are no replicas on " + diskMountPath + " of " + dataNodeId);
+    }
+    logger.info("Found {} on {}", replicasOnDisk, diskMountPath);
+
+    // move the last replica id to scratch space
+    ReplicaId toMove = replicasOnDisk.get(replicasOnDisk.size() - 1);
+    File scratchSrc = new File(toMove.getReplicaPath());
+    File scratchTgt = new File(scratch, TEMP_RELOCATION_DIR_NAME);
+    logger.info("Moving {} to {}", scratchSrc, scratchTgt);
+    delete(scratchTgt);
+    Files.move(scratchSrc.toPath(), scratchTgt.toPath());
+
+    // reformat each store, except the one moved, one by one
+    for (int i = 0; i < replicasOnDisk.size() - 1; i++) {
+      ReplicaId replicaId = replicasOnDisk.get(i);
+      File src = new File(replicaId.getReplicaPath());
+      File tgt = new File(replicaId.getMountPath(), TEMP_COPY_DIR_NAME);
+      logger.info("Copying {} to {}", src, tgt);
+      copy(src, tgt, replicaId.getCapacityInBytes());
+      delete(src);
+      if (!tgt.renameTo(src)) {
+        throw new IllegalStateException("Could not rename " + tgt + " to " + src);
+      }
+      logger.info("Done reformatting {}", replicaId);
+    }
+
+    // reformat the moved store
+    copy(scratchTgt, scratchSrc, toMove.getCapacityInBytes());
+    delete(scratchTgt);
+    logger.info("Done reformatting {}", toMove);
+  }
+
+  /**
+   * Copy the partition at {@code src} to {@code tgt} using a {@link StoreCopier}.
+   * @param src the location of the partition to be copied
+   * @param tgt the location where the partition has to be copied to
+   * @param capacityInBytes the capacity of the partition.
+   * @throws Exception
+   */
+  private void copy(File src, File tgt, long capacityInBytes) throws Exception {
```

Comment:
Unfortunately not. This is from `ConsistencyCheckerTool.checkConsistency()` which eventually uses `DumpIndexTool.dumpIndex()`. I did some fixing around to make them API-like in #707 but there is a still a lot of work to do there and perhaps that can eventually stop throwing `Exception`.